core/common/blob.hpp

```diff
@@ -74,13 +74,19 @@ namespace kagome::common {
   template <size_t size_>
   class Blob : public std::array<byte_t, size_> {
    public:
+    using const_narref = const byte_t (&)[size_];
+    using const_narptr = const byte_t (*)[size_];
     /**
      * Initialize blob value
      */
     Blob() {
       this->fill(0);
     }
 
+    const_narref internal_array_reference() const {
```

Comment:
I'm not sure, how it is more beneficial to introduce this method instead of casting data() in the place where it is used? It doesn't seem that this is going to be used often.

Reply:
```data``` returns a pointer without any information of array length. I gave this info for the compiler to make him able to decompose a loop in a place of processing. It was made to activate ```out-of-order execution```.